payment.yaml

```diff
@@ -0,0 +1,187 @@
+openapi: 3.0.0
+info:
+  version: 1.0.0
+  title: Payment Ecosystem
+  description: Internal API that supplies abstraction over blockchain
+servers:
+  - url: 'https://api.kinmarketplace.internal/v1/'
+paths:
+  /transactions/{transaction_id}:
+    parameters:
+      - $ref: '#/components/parameters/RequestId'
+      - $ref: '#/components/parameters/BlockingTime'
+      - in: path
+        name: transaction_id
+        schema:
+          type: string
+        required: true
+        description: blockchain transaction id
+    get:
+      tags:
+        - Transactions
+      description: get transaction info
+      operationId: getTransaction
+      responses:
+        '200':
+          description: Successfully returned transaction info
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Transaction'
+        '404':
+          description: Transaction not found
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+        default:
+          description: Generic Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+                
+  /wallets/{public_address}:
+    parameters:
+      - $ref: '#/components/parameters/RequestId'
+      - $ref: '#/components/parameters/BlockingTime'
+      - in: path
+        name: public_address
+        schema:
+          type: string
+        required: true
+        description: public address of wallet to create/ fund in native currency
+    post:
+      tags:
+        - Transactions
+
+      description: get transaction info
+      operationId: createWallet
+      responses:
+        '204':
+          description: Successfully created wallet
+        default:
+          description: Generic Error
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/Error'
+  
+  /wallets/{public_address}/transactions:
```

Comment:
`/wallets/{addr}` is for working with wallets, not with payments.
better POST to `/transactions` or `/payments`, this will create a new payment.

Reply:
removed this